src/App.tsx

```diff
@@ -15,49 +17,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: { sortType: SortType, isReversed: boolean },
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType) {
+    visibleGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const { NONE, ALPHABET, LENGTH } = SortType;
+  const [sortType, setSortType] = useState(NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const realGoods = getReorderedGoods(goodsFromServer, {
+    sortType,
+    isReversed,
+  });
+
+  const isSorted = sortType === ALPHABET || sortType === LENGTH || isReversed;
+
+  const handleReset = () => {
+    setSortType(NONE);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortType !== ALPHABET,
+          })}
+          onClick={() => setSortType(ALPHABET)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortType !== LENGTH,
+          })}
+          onClick={() => setSortType(LENGTH)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
In scenarios where the new state depends on the previous state, it is recommended to use the functional form of setState to ensure the accuracy of state updates.
```
setState(prevValue => !prevValue)